android/jni/com/mapswithme/platform/Platform.cpp

```diff
@@ -136,21 +136,9 @@ platform::NetworkPolicy GetCurrentNetworkPolicy()
 
 namespace android
 {
-Platform::Platform()
-{
-  auto env = jni::GetEnv();
```

Comment:
@biodranik do you have an idea why settings should be initialized in the constructor?

I see that it is actually broken after my path:

```
03-20 19:04:34.498 29871 29871 W MapsmeCore: platform/string_storage_base.cpp:69 Save()  Saving settings: settings.ini; Write truncate; Read-only file system settings.ini
03-20 19:04:34.544 29871 31359 W MapsmeCore: platform/string_storage_base.cpp:69 Save()  Saving settings: settings.ini; Write truncate; Read-only file system settings.ini
03-20 19:09:28.727 29871 29871 W MapsmeCore: platform/string_storage_base.cpp:69 Save()  Saving settings: settings.ini; Write truncate; Read-only file system settings.ini
03-20 19:09:29.423 29871 29871 W MapsmeCore: platform/string_storage_base.cpp:69 Save()  Saving settings: settings.ini; Write truncate; Read-only file system settings.ini
03-20 19:09:43.524 29871 29871 W MapsmeCore: platform/string_storage_base.cpp:69 Save()  Saving settings: settings.ini; Write truncate; Read-only file system settings.ini
```

Reply:
C++ Settings is a global static object which should be initialized as early as possible before other code starts using Settings:: methods.

Reply:
Yes, I realized that. See the updated version. In the new version I initialize m_settingsPath from MwmApplication.onCreate().